apiclient/rancher_api/managers.py

```diff
@@ -0,0 +1,190 @@
+        try:
+            return resp.status_code, resp.json()
+        except json.decoder.JSONDecodeError as e:
+            return resp.status_code, dict(error=e, response=resp)
+
+    def _get(self, path, *, raw=False, **kwargs):
+        return self._delegate("_get", path, raw=raw, **kwargs)
+
+    def _create(self, path, *, raw=False, **kwargs):
+        return self._delegate("_post", path, raw=raw, **kwargs)
+
+    def _update(self, path, data, *, raw=False, as_json=True, **kwargs):
+        if as_json:
+            kwargs.update(json=data)
+        else:
+            kwargs.update(data=data)
+
+        return self._delegate("_put", path, raw=raw, **kwargs)
+
+    def _delete(self, path, *, raw=False, **kwargs):
+        return self._delegate("_delete", path, raw=raw, **kwargs)
+
+
+class SettingManager(BaseManager):
+    # server-version
+    PATH_fmt = "apis/management.cattle.io/v3/settings/{name}"
+    # "v1/harvesterhci.io.settings/{name}"
+
+    def get(self, name="", *, raw=False):
+        return self._get(self.PATH_fmt.format(name=name))
+
+
+class MgmtClusterManager(BaseManager):
+    CREATE_fmt = "v1/provisioning.cattle.io.clusters"
+    GET_fmt = "apis/provisioning.cattle.io/v1/namespaces/{ns}/clusters/{uid}"
+    DELETE_fmt = "v1/provisioning.cattle.io.clusters/{ns}/{uid}"
+
+    def create_data(self, cluster_name, spec={}, labels={}):
+        return {
+            "type": "provisioning.cattle.io.cluster",
+            "metadata": {
+                "namespace": FLEET_DEFAULT_NAMESPACE,
+                "labels": labels,
+                "name": cluster_name
+            },
+            "spec": spec
+        }
+
+    def get(self, name="", *, raw=False):
+        return self._get(self.GET_fmt.format(uid=name, ns=FLEET_DEFAULT_NAMESPACE), raw=raw)
+
+    def create(self, name, spec={}, labels={}, raw=False):
+        data = self.create_data(name, spec=spec, labels=labels)
+        return self._create(self.CREATE_fmt, json=data, raw=raw)
+
+    def delete(self, name, *, raw=False):
+        return self._delete(self.DELETE_fmt.format(uid=name, ns=FLEET_DEFAULT_NAMESPACE))
+
+
+class ClusterRegistrationTokenManager(BaseManager):
+    PATH_fmt = "v3/clusterRegistrationTokens/{uid}:default-token"
+
+    def get(self, name="", *, raw=False):
+        return self._get(self.PATH_fmt.format(uid=name), raw=raw)
+
+
+class CloudCredentialManager(BaseManager):
+    PATH_fmt = "v3/cloudcredentials"
+
+    def create_data(self, name, cluster_id, kubeconfig):
+        return {
+            "type": "provisioning.cattle.io/cloud-credential",
+            "metadata": {
+                "generateName": "cc-",
+                "namespace": FLEET_DEFAULT_NAMESPACE
+            },
+            "_name": name,
+            "annotations": {
+                "provisioning.cattle.io/driver": "harvester"
+            },
+            "harvestercredentialConfig": {
+                "clusterType": "imported",
+                "clusterId": cluster_id,
```

Comment:
`clusterType` and `clusterId` should not be constant in here, we can create **Cloud Credential** by _External Harvester_, so `clusterType` should be determined by `clusterId` (if not provided, `clusterType` should be `external`)